tests/functionaltests/Tests/UIKitTests/UIButtonTests.mm

```diff
@@ -51,75 +36,108 @@
-                EXPECT_EQ_MSG(solidBrush.color.r, 127, @"Failed to match expected value for red component");
-                EXPECT_EQ_MSG(solidBrush.color.g, 127, @"Failed to match expected value for green component");
-                EXPECT_EQ_MSG(solidBrush.color.b, 127, @"Failed to match expected value for blue component");
-                EXPECT_EQ_MSG(solidBrush.color.a, 255, @"Failed to match expected value for alpha component");
-
-                // Unregister the callback
-                [backingElement unregisterPropertyChangedCallback:[WXCControl backgroundProperty] token:callbackToken];
-
-                [condition lock];
-                [condition signal];
-                [condition unlock];
-        }];
+TEST(UIButton, TitleColorChanged) {
+    __block BOOL signaled = NO;
+    __block NSCondition* condition = [[NSCondition alloc] init];
+    __block UXTestAPI::XamlEventSubscription xamlSubscriber;
+
+    UIButtonWithControlsViewController* buttonVC = [[UIButtonWithControlsViewController alloc] init];
+    UXTestAPI::ViewControllerPresenter testHelper(buttonVC);
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        // Extract UIButton.titleLabel control to verify its visual state
+        WXFrameworkElement* titleElement = [buttonVC.button.titleLabel xamlElement];
+        ASSERT_TRUE(titleElement);
+
+        // Register RAII event subscription handler
+        xamlSubscriber.Set(titleElement, [WXCTextBlock foregroundProperty], ^(WXDependencyObject* sender, WXDependencyProperty* dp) {
+            // Validate the state of the XAML object
+            WUXMSolidColorBrush* solidBrush = rt_dynamic_cast([WUXMSolidColorBrush class], [sender getValue:dp]);
+            LOG_INFO("XAML element color (rgba): %d,%d,%d,%d",
+                     [solidBrush.color r],
+                     [solidBrush.color g],
+                     [solidBrush.color b],
+                     [solidBrush.color a]);
+
+            // Extract the title color for the normal state
+            UIColor* titleColorNormal = [buttonVC.button titleColorForState:UIControlStateNormal];
+
+            CGFloat red, green, blue, alpha;
+            [titleColorNormal getRed:&red green:&green blue:&blue alpha:&alpha];
+            LOG_INFO("UIButton.titleColorForState:normal (rgba): %.2f,%.2f,%.2f,%.2f", red, green, blue, alpha);
+
+            // Validate that the change is reflected on the backing XAML control
+            EXPECT_EQ_MSG(solidBrush.color.r, (int)(red * 255), @"Failed to match red component");
+            EXPECT_EQ_MSG(solidBrush.color.g, (int)(green * 255), @"Failed to match green component");
+            EXPECT_EQ_MSG(solidBrush.color.b, (int)(blue * 255), @"Failed to match blue component");
+            EXPECT_EQ_MSG(solidBrush.color.a, (int)(alpha * 255), @"Failed to match alpha component");
+
+            // Ensure that the XAML color values match the viewcontroller property
```

Comment:
We care about the actual color - the pattern in the future might be that we set this property explicitly (e.g. blueColor) Our verification should include that (XAML == UIButton.titleColor) and that both == blueColor.

---
In reply to: [97702035](https://github.com/Microsoft/WinObjC/pull/1771#discussion_r97702035) [](ancestors = 97702035)